Sample_v3/Samples/SwiftUISimpleChat/ChatView.swift

```diff
@@ -7,38 +7,111 @@ import SwiftUI
+        }
+        
+        return text
+            /// Inverted list workaround
             .scaleEffect(x: 1, y: -1, anchor: .center)
-            .offset(x: 0, y: 2)
-            
-            HStack {
-                TextField("Type a message", text: $text)
-                Button(action: self.send) {
-                    Text("Send")
+            /// Load next more messages when the last is shown
+            .onAppear {
+                if (self.channel.messages.last == message) {
+                    self.channel.controller.loadPreviousMessages()
                 }
-            }.padding()
-        }
-        .navigationBarTitle(Text(channelId.id), displayMode: .inline)
-        .onAppear(perform: onAppear)
+            }
+    }
+    
+    func composerView() -> some View {
+        HStack {
+            TextField("Type a message", text: $text)
+            Button(action: self.send) {
+                Image(systemName: "arrow.up.circle.fill")
+            }
+        }.padding()
     }
     
     func send() {
-        // send message
-        messages += [text]
+        guard let channelId = channel.channel?.cid else {
+            return
+        }
+        
+        if let editingMessage = self.editingMessage {
+            let controller = channel.controller.client.messageController(cid: channelId, messageId: editingMessage.id)
+            controller.editMessage(text: text)
+            self.editingMessage = nil
+        } else {
+            channel.controller.createNewMessage(text: text)
+        }
+
         text = ""
     }
     
-    func onAppear() {
-        // load messages
-        messages = ["Hello World!", "This is a test", "Mock message"]
+    /// Action sheet with channel actions
+    func actionSheet() -> ActionSheet {
+        if let message = editingMessage {
+            let messageController = channel.controller.client.messageController(cid: channel.channel!.cid, messageId: message.id)
+            return ActionSheet(title: Text("Message Actions"), message: Text(""), buttons: [
+                .default(Text("Edit"), action: { self.editingMessage = message; self.text = message.text }),
+                .destructive(Text("Delete")) { messageController.deleteMessage(); self.editingMessage = nil },
+                .cancel { self.editingMessage = nil }
+            ])
+        } else {
+            let userIds = Set(["steep-moon-9"])
+            return ActionSheet(title: Text("Channel Actions"), message: Text(""), buttons: [
+                .default(Text("Add Member"), action: { self.channel.controller.addMembers(userIds: userIds) }),
+                .default(Text("Remove Member"), action: { self.channel.controller.removeMembers(userIds: userIds) }),
+                .cancel()
+            ])
+        }
     }
 }
+
+/// Need to conform to Indentifiable for action sheet
+extension Message: Identifiable {}
```

Comment:
Can you please move this to our SDK? The same way we do it for `ChannelModel`

Reply:
@VojtaStavik I see that conformance is not needed anymore after I made a change to the action sheet logic (to support action sheet for messages and the channel at the same time, the trigger was changed from Message to a Bool). So I'm removing for now.